aws_lambda_powertools/logging/formatter.py

```diff
@@ -111,9 +111,15 @@ def __init__(
             Key-value to be included in log messages
 
         """
+
         self.json_deserializer = json_deserializer or json.loads
         self.json_default = json_default or str
-        self.json_serializer = json_serializer or partial(json.dumps, default=self.json_default, separators=(",", ":"))
+        self.json_indent = (
+            4 if os.getenv("AWS_SAM_LOCAL", "").lower() == "true" else None
```

Comment:
Suggestion: Use a constant to ease maintaining this later, e.g.

```suggestion
            PRETTY_INDENT if os.getenv("AWS_SAM_LOCAL", "").lower() == "true" else COMPACT_INDENT
```